docs/ocean/design_docs/vert-lagrang-remap/vert-lagrang-remap.rst

```diff
@@ -0,0 +1,124 @@
+
+Vertical Lagrangian-remap method
+======================
+
+date: 2020/11/24
+
+Contributors: Darren Engwirda, Xylar Asay-Davis, Carolyn Begeman
+
+
+
+Summary
+-------
+
+The vertical Lagrangian-remap method is to be implemented to support the choice 
+of hybrid vertical coordinates. It can be seen as an extension of the Arbitrary 
+Lagrangian-Eulerian method which is already implemented in MPAS-Ocean, a key 
+difference being that the solution of diasurface transport is coordinate-free. 
+This has the advantage of removing the vertical CFL constraint, which is of 
+particular concern for an evolving ice shelf geometry in terrain-following 
+coordinate schemes. Generalization of the vertical coordinate in the horizontal 
+equations of motion is not in the scope of this development. This development 
+will include some infrastructure for specifying the target grid for regridding. 
+The minimum criteria for success are not degrading the accuracy of the solution 
+or the performance by more than [an acceptable threshold] for several basic test 
+cases... .
+
+*The purpose of this section is to summarize what capability is to be added to
+MPAS-Model through this design process. It should be clear what new code will do
+that the current code does not. Summarizing the primary challenges with respect
+to software design and implementation is also appropriate for this section.
+Finally, this statement should contain a general statement with regard to what
+is "success."*
+
+
+Requirements
+------------
+
+Requirement: name-of-requirement-here
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+Date last modified: 2020/11/24
+
+Contributors: Darren Engwirda, Xylar Asay-Davis, Carolyn Begeman
+
+Requirement: A grid that relaxes to an analytical expression for a target grid.
+
+Requirement: The ability to implement hybrid coordinate systems.
+
+Requirement: XX order accuracy of the solution.
+
+Requirement: Performance is not degraded below an acceptable level.
+
+Requirement: Layer thicknesses are non-negative (and above a minimum thickness).
+
+*Possible additional requirements:
+- re. development of instabilities
+- re. conservation/monotonicity of interpolants
+*
```

Comment:
Your list are good requirements that should be fleshed out in full paragraphs if possible.

To me, some main requirements are:
* No calculations are made outside of remapping itself using prognostic (or diagnostic) variables both before and after remapping.  This is because the vertical locations of information before and after remapping may be completely unrelated to one another (separated by many layers).  Once we have a variable number of layers, this information could go from being "valid" to "invalid" or visa versa after remapping.
* Remapping should be supported (eventually) in both the RK4 and the split-explicit time stepping schemes.  For performance reasons, the sooner it can be supported in split-explicit, the better.  (In the design below, we may state that we will start with RK4.)
* To the extent that it is feasible, the user should have the ability to control the order of accuracy and other parameters of the remapping (e.g. via namelist options).
* To the degree possible, the code for performing remapping should be kept in its own Fortran module(s) for better readability.

Happy to keep brainstorming